src/Init/Data/List/Basic.lean

```diff
@@ -477,19 +574,33 @@ theorem reverseAux_reverseAux (as bs cs : List α) : reverseAux (reverseAux as b
 /-! ### append -/
 
 /--
-`O(|xs|)`: append two lists. `[1, 2, 3] ++ [4, 5] = [1, 2, 3, 4, 5]`.
-It takes time proportional to the first list.
+Appends two lists. Normally used via the `++` operator.
+
+Appending lists takes time proportional to the length of the first list: `O(|xs|)`.
+
+Examples:
+  * `[1, 2, 3] ++ [4, 5] = [1, 2, 3, 4, 5]`.
+  * `[] ++ [4, 5] = [4, 5]`.
+  * `[1, 2, 3] ++ [] = [1, 2, 3]`.
 -/
 protected def append : (xs ys : List α) → List α
   | [],    bs => bs
   | a::as, bs => a :: List.append as bs
 
-/-- Tail-recursive version of `List.append`.
+/--
+Appends two lists. Normally used via the `++` operator.
+
+Appending lists takes time proportional to the length of the first list: `O(|xs|)`.
 
-Most of the tail-recursive implementations are in `Init.Data.List.Impl`,
-but `appendTR` must be set up immediately,
-because otherwise `Append (List α)` instance below will not use it.
+This is a tail-recursive version of `List.append`.
+
+Examples:
+  * `[1, 2, 3] ++ [4, 5] = [1, 2, 3, 4, 5]`.
+  * `[] ++ [4, 5] = [4, 5]`.
+  * `[1, 2, 3] ++ [] = [1, 2, 3]`.
 -/
+-- Most of the tail-recursive implementations are in `Init.Data.List.Impl`, but `appendTR` must be
```

Comment:
Ooh, this is not actually true anymore. As many as possible of the TR implementations are now in this file, just the ones that use arrays in their implementations are in `List.Impl`.

Reply:
Comment updated! I got rid of the first part and kept the second, which seemed to still be relevant to have there.